### PR TITLE
Adds max/avg distribution plots

### PR DESCRIPTION
**Pull request by vulder**

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- varats/data/reports/blame_report.py  2
- varats/utils/git_util.py  3
         

Clones added
============
- varats/plots/blame_interaction_degree.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4678130&bid=15650871)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/utils/git_util.py  3
- varats/plots/blame_interaction_degree.py  1
- varats/data/reports/blame_report.py  2
         

Clones added
============
- varats/plots/blame_interaction_degree.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4678130&bid=15650871)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/data/reports/blame_report.py  2
- varats/utils/git_util.py  3
- varats/plots/blame_interaction_degree.py  1
         

Clones added
============
- varats/plots/blame_interaction_degree.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4678130&bid=15650871)

File: varats/data/reports/blame_report.py [hunk 1]
```diff
@@ -6,12 +6,15 @@
...
 from collections import defaultdict
 import yaml
+from datetime import datetime
+import numpy as np
 
 import pygit2
 
 from varats.data.report import BaseReport, MetaReport, FileStatusExtension
 from varats.data.version_header import VersionHeader
 from varats.utils.project_util import get_local_project_git
```
Comment (boehmseb) [inline varats/data/reports/blame_report.py]:
```suggestion
```

File: varats/utils/git_util.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+
+MC_RET = tp.TypeVar("MC_RET")
+
+
+def map_commits(func: tp.Callable[[pygit2.Commit], MC_RET],
+                c_hash_list: tp.Iterable[str],
+                commit_lookup: tp.Callable[[str], pygit2.Commit]
+               ) -> tp.Sequence[MC_RET]:
+    # Skip 0000 hashes that we added to mark uncommited files
```
Comment (boehmseb) [inline varats/utils/git_util.py]:
```suggestion
    # Skip 0000 hashes that we added to mark uncommitted files
```

**Review by boehmseb — CHANGES_REQUESTED** [review]

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/data/reports/blame_report.py  2
- varats/utils/git_util.py  3
- varats/plots/blame_interaction_degree.py  1
         

Clones added
============
- varats/plots/blame_interaction_degree.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4678130&bid=15650871)

**Pull request merged**